### PR TITLE
Resolving mysql deprecated operator warnings

### PR DESCRIPTION
**Pull request by dirrao**
Resolving mysql deprecated operator warnings

related: #39485

### CODE REVIEW

**Comment by Bowrna** [comment]
@dirrao you may need to edit the documentation part too
airflow/docs/apache-airflow-providers-mysql/operators.rst

**Review by Taragolis — APPROVED** [review]

File: airflow/providers/mysql/operators/mysql.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class MySqlOperator(SQLExecuteQueryOperator):
 
     .. seealso::
         For more information on how to use this operator, take a look at the guide:
-        :ref:`howto/operator:MySqlOperator`
+        :ref:`howto/operator:mysql`
```
Comment (Taragolis) [inline airflow/providers/mysql/operators/mysql.py:45]: Do we really need to change target from `MySqlOperator` to generic `mysql` ?

Comment (dirrao) [reply airflow/providers/mysql/operators/mysql.py:45]: Yes. The doc checks are failing as this reference is not available.

**Comment by dirrao** [comment]
@eladkal / @potiuk 
Can someone check and merge this pull request?

**Review by potiuk — APPROVED** [review]

**Pull request merged**